### ISSUE TITLE
MultipartReader instance's readForm function caused socket hang up

### ISSUE THREAD

**Issue by lengfangbing** [denoland/deno#5866]
I use MultipartReader to read multipart/form-data body. Thats ok, but when I use readForm function, It got a strange question.
The first request, it worked.

![image](https://user-images.githubusercontent.com/46396510/82883826-8cc60880-9f75-11ea-84e4-de16eed13db5.png)

But the next request, it showed socket hang up error.

![image](https://user-images.githubusercontent.com/46396510/82883901-a2d3c900-9f75-11ea-9e09-189f524ca52c.png)

Go ahead, the next was ok, then the next was the same error. repeat again and again...

here is code. You can find these code in middleWare/requestBody/requestBody.ts
[deno-demo.zip](https://github.com/denoland/deno/files/4681187/deno-demo.zip)

**Comment by devYusuf** [comment]
How did you parse the form data first, iv'e been trying to get this for days

**Comment by lengfangbing** [comment]
> How did you parse the form data first, iv'e been trying to get this for days

In oak. 
```javascript
app.use(async (ctx: Context, next: Function) => {
    const originRequest = ctx.request.serverRequest;
    const contentType = ctx.request.headers.get('content-type') || 'text';
    const body = await parseFormData(contentType, originRequest);
    ctx.body = body;
    await next();
});
// parse multipart data
async function parseFormData(contentType: string, request: ServerRequest) {
  const boundaryReg = /boundary=(.+)/i;
  const match = contentType.match(boundaryReg);
  const boundary = match?.[1];
  const res: {[key: string]: string | FormFile} = {};
  if(boundary) {
    try {
      const mr = new MultipartReader(request.r, boundary);
      const form = await mr.readForm(2 * 1024 * 1024);
      if (form) {
        for (const [key, value] of form.entries()) {
          const newValue: any = value || {};
          const val = <FormFile>value;
          if (val.content) {
            newValue.value = decoder.decode(val.content);
          }
          res[key] = <string | FormFile>value;
        }
      }
      return res;
    } catch (e) {
      console.log(e);
      throw e;
    }
  }
}
```
But as I mentioned. I got some strange error. It may caused by readForm. I am confusing now X.X

**Comment by lengfangbing** [comment]
I tested again. I want to share more information so we can solve this.
Here, I used readForm.

![image](https://user-images.githubusercontent.com/46396510/82904213-b9d5e380-9f94-11ea-8f61-a9cdb8bcf17e.png)

I use both fetch and xhr to test it.
The result here.

![image](https://user-images.githubusercontent.com/46396510/82904542-336dd180-9f95-11ea-9fad-222427c64729.png)

It showed that failed by cors policy. But If it's blocked by cors policy, why first time sent successful X.X

Then I delete the readForm function.

![image](https://user-images.githubusercontent.com/46396510/82904647-539d9080-9f95-11ea-9269-cd8a3ec7fb7b.png)

The result here.

![image](https://user-images.githubusercontent.com/46396510/82904719-6e700500-9f95-11ea-9056-e82d8e7686c3.png)

all the request was successful...
